pkg/bloomgateway/bloomgateway.go

```diff
@@ -415,10 +398,41 @@ outer:
 	g.metrics.addUnfilteredCount(numChunksUnfiltered)
 	g.metrics.addFilteredCount(len(req.Refs))
 
-	level.Debug(g.logger).Log("msg", "return filtered chunk refs", "unfiltered", numChunksUnfiltered, "filtered", len(req.Refs))
+	level.Info(logger).Log("msg", "return filtered chunk refs", "unfiltered", numChunksUnfiltered, "filtered", len(req.Refs))
 	return &logproto.FilterChunkRefResponse{ChunkRefs: req.Refs}, nil
 }
 
+// consumeTask receives v1.Output yielded from the block querier on the task's
+// result channel and stores them on the task.
+// In case the context task is done, it drains the remaining items until the
+// task is closed by the worker.
+// Once the tasks is closed, it will send the task with the results from the
+// block querier to the supplied task channel.
+func consumeTask(ctx context.Context, task Task, tasksCh chan<- Task, logger log.Logger) {
+	logger = log.With(logger, "task", task.ID)
+
+	task.responses = responsesPool.Get(len(task.series))
+	defer responsesPool.Put(task.responses)
```

Comment:
We are returning the task (and therefore its responses), but we are also putting back the responses to the pool. I think this may cause a race condition where another task can overwrite the values of the responses.

Reply:
Good point. Gonna remove the pool.